system/Model.php

```diff
@@ -788,17 +788,32 @@ protected function objectToRawArray($data, bool $onlyChanged = true, bool $recur
     {
         $properties = parent::objectToRawArray($data, $onlyChanged);
 
+        $primaryKey = null;
+
+        // For Entity
+        if (method_exists($data, 'cast')) {
```

Comment:
This seems a very loose check. Is there precedence for this? What is wrong with `instanceof Entity`? I would even be in favor of adding `interface Castable` over this.

Reply:
Exactly! The check is not good. Fixed.

I think `interface Castable` is not needed.
It is an Entity-specific process and I don't think it should be generalized.